### PR TITLE
Promotion OptionValue Rules

### PR DESCRIPTION
**Pull request by philbirt**
I believe this is something that was meant to be brought over from our fork, but somehow we missed a commit. This is just a cherry-pick and cleanup of https://github.com/bonobos/spree/commit/e7813d81265548b06970339fcad35dfa5f906acd

We currently have a good portion of the code in solidus that does the heavy-lifting, but it is not visible in the admin UI. This will get it visible and working.

### CODE REVIEW

**Comment by magnusvk** [comment]
@philbirt That spec failure looks real, right?

**Comment by philbirt** [comment]
Unfortunately yeah. I had gotten it working with selenium but it doesn't seem to work with poltergeist. I'm not sure why yet.

**Comment by philbirt** [comment]
@magnusvk @adammathys @jhawthorn The feature spec is now passing, would you gentlemen mind taking another look?

File: backend/app/assets/javascripts/spree/backend/promotions.js.coffee [hunk 1]
```diff
@@ -26,6 +26,44 @@ window.initProductActions = ->
         $settings.find('input').prop 'disabled', 'disabled'
 
   #
+  # Option Value Promo Rule
+  #
+  if $('#promo-rule-option-value-template').length
+    optionValueSelectNameTemplate = Handlebars.compile($('#promo-rule-option-value-option-values-select-name-template').html())
```
Comment (magnusvk) [inline backend/app/assets/javascripts/spree/backend/promotions.js.coffee]: We should move these to precompiled Handlebars templates like https://github.com/solidusio/solidus/commit/e757bdb04649d0068c848a29376d83baafc3f0b0 did.

**Comment by magnusvk** [comment]
Looks great otherwise.

**Comment by jhawthorn** [comment]
Other than precompiling handlebars looks good here

**Comment by magnusvk** [comment]
Thanks! :+1:

**Pull request merged**